opm/parser/eclipse/Deck/Section.hpp

```diff
@@ -47,7 +47,7 @@ class Section : public DeckView {
 
         // returns whether the deck has all mandatory sections and if all sections are in
         // the right order
-        static bool checkSectionTopology(const Deck& deck,
```

Comment:
>The member variable must be marked mutable,

Yes, that in ``Deck.hpp`` ``mutable MessageContainer m_messages`` is made,  the error happened in ``checkDeck.hpp``.

``
diff --git a/opm/parser/eclipse/Deck/Deck.hpp b/opm/parser/eclipse/Deck/Deck.hpp
index a4e9815..99b0fb3 100644
--- a/opm/parser/eclipse/Deck/Deck.hpp
+++ b/opm/parser/eclipse/Deck/Deck.hpp
@@ -121,7 +121,7 @@ namespace Opm {
             void initUnitSystem() const;
 
             std::vector< DeckKeyword > keywordList;
-            MessageContainer m_messageContainer;
+            mutable MessageContainer m_messageContainer;
 
             mutable std::unique_ptr< UnitSystem > activeUnits;
             mutable std::unique_ptr< UnitSystem > defaultUnits;
diff --git a/opm/parser/eclipse/Deck/Section.hpp b/opm/parser/eclipse/Deck/Section.hpp
index 0e85143..a21f4b5 100644
--- a/opm/parser/eclipse/Deck/Section.hpp
+++ b/opm/parser/eclipse/Deck/Section.hpp
@@ -47,7 +47,7 @@ class Section : public DeckView {
 
         // returns whether the deck has all mandatory sections and if all sections are in
         // the right order
-        static bool checkSectionTopology(Deck& deck,
+        static bool checkSectionTopology(const Deck& deck,
                                          const Parser&,
                                          bool ensureKeywordSectionAffiliation = false);
``

Error:
``
In function ‘bool Opm::checkDeck(Opm::DeckConstPtr, Opm::ParserConstPtr, size_t)’:
EclipseState/checkDeck.cpp:36:43: error: passing ‘const Opm::Deck’ as ‘this’ argument of ‘Opm::MessageContainer& Opm::Deck::getMessageContainer()’ discards qualifiers [-fpermissive]
                 deck->getMessageContainer().warning(keyword.getFileName(), msg,keyword.getLineNumber());
                                           ^
``